tests/Doctrine/Tests/DBAL/Platforms/AbstractPostgreSqlPlatformTestCase.php

```diff
@@ -819,6 +819,29 @@ public function testAltersTableColumnCommentWithExplicitlyQuotedIdentifiers()
         );
     }
 
+    /**
+     * @group 3158
+     */
+    public function testAltersTableColumnCommentIfRequiredByType()
+    {
+        $table1 = new Table('"foo"', [new Column('"bar"', Type::getType('datetime'))]);
+        $table2 = new Table('"foo"', [new Column('"bar"', Type::getType('datetime_immutable'))]);
+
+        $comparator = new Comparator();
+
+        $tableDiff = $comparator->diffTable($table1, $table2);
+
+        $this->assertInstanceOf('Doctrine\DBAL\Schema\TableDiff', $tableDiff);
+        $this->assertSame(
+            [
+                'ALTER TABLE "foo" ALTER "bar" TYPE TIMESTAMP(0) WITHOUT TIME ZONE',
+                'ALTER TABLE "foo" ALTER "bar" DROP DEFAULT',
+                'COMMENT ON COLUMN "foo"."bar" IS \'(DC2Type:datetime_immutable)\'',
```

Comment:
Does the patch only affect types that require comments, or comments in general? If the latter is the case, can it be generalised to use explicit comments rather than going through DBAL types?

Reply:
only when the type requires a comment via `requiresSQLCommentHint` as far as i can see. General comments (at least those added via `options={"comment":"..."}` to an orm column are correctly picked up.

Reply:
secondly, there is a further 'bug' here. the first two sql statements are not required as they are the same between the types being changed. I can fix this in a separate PR rather than pollute this one. they won't cause a problem.